pynestml/codegeneration/nest_builder.py

```diff
@@ -0,0 +1,124 @@
+
+            nest_path = nest.ll_api.sli_func("statusdict/prefix ::")
+            self.set_options({"nest_path": nest_path})
+            Logger.log_message(None, -1, "The NEST installation was automatically detected as: " + nest_path, None, LoggingLevel.INFO)
+
+    def build(self) -> None:
+        r"""
+        This method can be used to build the generated code and install the resulting extension module into NEST.
+
+        Raises
+        ------
+        GeneratedCodeBuildException
+            If any kind of failure occurs during cmake configuration, build, or install.
+        InvalidPathException
+            If a failure occurs while trying to access the target path or the NEST installation path.
+        """
+        cmake_cmd = ["cmake"]
+        target_path = FrontendConfiguration.get_target_path()
+        install_path = FrontendConfiguration.get_install_path()
+        if install_path is not None:
+            add_libraries_to_sli(install_path)
+        nest_path = self.get_option("nest_path")
+
+        if not os.path.isdir(target_path):
+            raise InvalidPathException('Target path (' + target_path + ') is not a directory!')
+
+        if nest_path is None or (not os.path.isdir(nest_path)):
+            raise InvalidPathException('NEST path (' + str(nest_path) + ') is not a directory!')
+
+        install_prefix = ""
+        if install_path:
+            if not os.path.isabs(install_path):
+                install_path = os.path.abspath(install_path)
+            install_prefix = f"-DCMAKE_INSTALL_PREFIX={install_path}"
+
+        nest_config_path = f"-Dwith-nest={os.path.join(nest_path, 'bin', 'nest-config')}"
+        cmake_cmd = ['cmake', nest_config_path, install_prefix, '.']
+        make_all_cmd = ['make', 'all']
+        make_install_cmd = ['make', 'install']
+
+        # remove CMakeCache.txt if exists
+        cmake_cache = os.path.join(target_path, "CMakeCache.txt")
+        if os.path.exists(cmake_cache):
+            os.remove(cmake_cache)
+
+        # check if we run on win
+        if sys.platform.startswith('win'):
+            shell = True
+        else:
+            shell = False
+
+        # first call cmake with all the arguments
+        try:
+            result = subprocess.check_call(cmake_cmd, stderr=subprocess.STDOUT, shell=shell,
+                                           cwd=str(os.path.join(target_path)))
+        except subprocess.CalledProcessError as e:
+            raise GeneratedCodeBuildException('Error occurred during \'cmake\'! More detailed error messages can be found in stdout.')
+
+        # now execute make all
+        try:
+            subprocess.check_call(make_all_cmd, stderr=subprocess.STDOUT, shell=shell,
+                                  cwd=str(os.path.join(target_path)))
+        except subprocess.CalledProcessError as e:
+            raise GeneratedCodeBuildException('Error occurred during \'make all\'! More detailed error messages can be found in stdout.')
```

Comment:
I thought the PR containing the `add_libraries_to_sli` has modified the exceptions:
```suggestion
            msg = "Error during 'make all'. More detailed error messages can be found in stdout."
            raise GeneratedCodeBuildException(msg)
```